src/pal/src/misc/sysinfo.cpp

```diff
@@ -346,7 +346,7 @@ PAL_GetLogicalProcessorCacheSizeFromOS()
 {
     size_t cacheSize = 0;
 
-#if HAVE_SYSCONF && defined(__linux__)
+#if HAVE_SYSCONF && defined(__linux__) && defined(_SC_LEVEL1_DCACHE_SIZE)
```

Comment:
I do think it needs to have `__Linux__` and `HAVE_SYSCONF`. It should be cleaner to just do:
```
#ifdef _SC_LEVEL1_DCACHE_SIZE
   cacheSize = max(cacheSize, sysconf(_SC_LEVEL1_DCACHE_SIZE)); 
#endif
#ifdef _SC_LEVEL2_CACHE_SIZE
...
#endif
```

While you are on it, you can also omit the `_SC_LEVEL1_ICACHE_SIZE` case. This API is meant to return size of the data caches for GC. It does not make a lot of sense to incorporate the size of instruction cache in the result.